aopy/analysis/base.py

```diff
@@ -2372,4 +2376,261 @@ def simulate_ideal_trajectories(targets, origin=[0.0, 0.0, 0.0], resolution=1000
         traj = np.linspace(origin, target, num_points)
         trajectories.append(traj)
         
-    return np.array(trajectories)
\ No newline at end of file
+    return np.array(trajectories)
+
+def calc_statistic_random_trials(data, n_trials=300, statistic=partial(np.mean, axis=0)):
+    '''
+    Calculate a distribution of a statistic across groups of trials of the data.
+
+    Args:
+        data (ntrials, nch): data to calculate the statistic on
+        n_trials (int): number of trials to use in each bootstrap
+        statistic (function): function to calculate the statistic on the data. Should
+            take the form `f(x) = y` where `x` is a 2D array of shape (n_trials, nch)
+            and `y` is a 1D array of shape (nch). Default is np.mean(x, axis=0).
+    
+    Returns:
+        (len(data)//ntrials, nch) array: distributions of the statistic across divisions of the data.
+
+    Examples:
+
+        .. code-block:: python
+
+            elec_pos, _, _ = aopy.data.load_chmap()
+            n_elec = len(elec_pos)
+            total_trials = 1600
+            n_trials = 50
+            np.random.seed()
+            data = 0.5*np.ones((total_trials,n_elec))
+            data += np.random.normal(0.5, size=(total_trials,n_elec))
+            data[:,n_elec//4:n_elec//2] += 0.1
+
+            dists = aopy.analysis.calc_statistic_random_trials(data, n_trials=n_trials)
+            self.assertEqual(np.shape(dists), (len(data)//n_trials, n_elec))
+
+            # Test that the distribution means are close to the original data mean
+            mean = np.mean(data, axis=0)
+            dists_mean = np.mean(dists, axis=0)
+            dists_std = np.std(dists, axis=0)
+            for i in range(n_elec):
+                self.assertAlmostEqual(mean[i], dists_mean[i], delta=0.1)
+            clim = (np.min(dists_mean), np.max(dists_mean))
+
+            plt.figure(figsize=(5,2), dpi=300)
+            plt.subplot(1,3,1)
+            im = aopy.visualization.plot_spatial_drive_map(mean, 
+                                            elec_data=True, cmap='Grays')
+            im.set_clim(*clim)
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('data mean')
+
+            plt.subplot(1,3,2)
+            im = aopy.visualization.plot_spatial_drive_map(dists_mean, 
+                                            elec_data=True, cmap='Grays')
+            im.set_clim(*clim)
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist means')
+
+            plt.subplot(1,3,3)
+            im = aopy.visualization.plot_spatial_drive_map(dists_std, 
+                                            elec_data=True, cmap='Grays')
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist std')
+
+        .. image:: _images/calc_statistic_random_trials.png
+    '''
+    all_trials = np.arange(len(data))
+    np.random.shuffle(all_trials)
```

Comment:
Would it be helpful to add an input arg that defines the seed? Or if you do that separately will it propagate when this function is called?

Reply:
good idea, actually this is a bug when run in parallel. fixed now.